src/kudu/consensus/leader_election.cc

```diff
@@ -138,6 +138,113 @@ bool VoteCounter::AreAllVotesIn() const {
   return GetTotalVotesCounted() == num_voters_;
 }
 
+void FlexibleVoteCounter::FetchTopologyInfo(
+    std::map<std::string, int>& voter_distribution,
+    std::map<std::string, int>& data_quorum_requirement) {
+  voter_distribution.clear();
+  data_quorum_requirement.clear();
+  // TODO(ritwikyadav): Populate the out params by reading config.
+}
+
+FlexibleVoteCounter::FlexibleVoteCounter(RaftConfigPB config)
+  : VoteCounter(0, 0),
+    config_(std::move(config)) {
+  num_voters_ = 0;
+  yes_vote_count_.clear();
+  no_vote_count_.clear();
+  FetchTopologyInfo(voter_distribution_, data_quorum_requirement_);
+  for (auto regional_vote_count : voter_distribution_) {
```

Comment:
nit: it seems you are iterating over voters, not votes. better to name it regional_voter_count